idemixmsp.go

```diff
@@ -82,11 +84,37 @@ var mspLogger = flogging.MustGetLogger("idemix")
 var mspIdentityLogger = flogging.MustGetLogger("idemix.identity")
 
 // NewIdemixMsp creates a new instance of idemixmsp
+// with the default curve of FP256BN_AMCL
 func NewIdemixMsp(version MSPVersion) (MSP, error) {
+	mspLogger.Debugf("Creating Idemix-based MSP instance with default curve ")
+	return NewIdemixMspCurve(version, math.FP256BN_AMCL)
+}
+
+// NewIdemixMspCurve creates a new instance of idemixmsp allowing the Elliptic Curve to be specified
+// Must be one of the support curves from https://github.com/IBM/mathlib
+func NewIdemixMspCurve(version MSPVersion, curveID math.CurveID) (MSP, error) {
 	mspLogger.Debugf("Creating Idemix-based MSP instance")
 
-	curve := math.Curves[math.FP256BN_AMCL]
-	csp, err := idemix.New(&keystore.Dummy{}, curve, &amcl.Fp256bn{C: curve}, true)
+	var curve *math.Curve
+	var tr idemix2.Translator
+
+	// as per the idemixgen file determine which
+	// curve is being used, and proceed accordingly
+	switch curveID {
+	case math.FP256BN_AMCL:
```

Comment:
Considered that it might be better to move some of this to the Mathlib library... i.e. be able to get from the mathlib library the 'translator' structure - save having to manually code it here.

Reply:
Need to add some more specific tests as well...